dashboard/app/models/section.rb

```diff
@@ -77,8 +77,8 @@ def name_safe_students
     # normalization, this is necessary to ensure that we can recognize
     # the name on the other side
     self.students.each do |student|
-      normalized = student.name.split(/\s+/).join(' ')
-      trie.add normalized
+      student.name = student.name.strip.split(/\s+/).join(' ')
```

Comment:
Another option might be to `.to_a` on line 79 so we're working on a copy of the data?  Or `student.merge(name: normalized)` on line 130?